pkg/karmadactl/deinit.go

```diff
@@ -0,0 +1,345 @@
+package karmadactl
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/utils"
+)
+
+// LabelSelector  karmada bootstrapping label
+const LabelSelector = "karmada.io/bootstrapping"
+
+// CommandDeInitOption  options for deinit.
+type CommandDeInitOption struct {
+	Recursive       bool
+	Force           bool
+	KarmadaDataPath string
+	Namespace       string
+	KubeConfig      string
+	KubeClientSet   *kubernetes.Clientset
+}
+
+// NewCmdDeInit remove karmada on kubernetes
+func NewCmdDeInit(cmdOut io.Writer, parentCommand string) *cobra.Command {
+	opts := CommandDeInitOption{}
+	cmd := &cobra.Command{
+		Use:          "deinit",
+		Short:        "remove karmada on kubernetes",
+		Long:         "remove karmada on kubernetes",
+		Example:      deInitExample(parentCommand),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			if err := opts.Complete(); err != nil {
+				return err
+			}
+			if err := opts.RunDeInit(); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	flags := cmd.PersistentFlags()
+	flags.StringVarP(&opts.Namespace, "namespace", "n", "karmada-system", "Kubernetes namespace")
+	flags.StringVar(&opts.KubeConfig, "kubeconfig", filepath.Join(homeDir(), ".kube", "config"), "absolute path to the kubeconfig file")
+	flags.StringVarP(&opts.KarmadaDataPath, "karmada-data", "d", "/etc/karmada", "karmada data path. kubeconfig cert and crds files")
+	flags.BoolVarP(&opts.Force, "force", "f", false, "skip delete confirmation")
+	flags.BoolVarP(&opts.Recursive, "recursive", "r", false, "delete recursively, delete all resources in the namespace, including persistent storage.")
+	return cmd
+}
+func deInitExample(parentCommand string) string {
+	return fmt.Sprintf(`
+# Remove Karmada in Kubernetes cluster.
+%s deinit
+
+# Skip the prompt and just delete Karmada.
+%s deinit --force
```

Comment:
Seems the `--force` flag is usually used to force do something not allowed by default. Not to ignore the prompt.

No better suggestion for now.

Reply:
In my opinion, we don't need `prompt` that seems rarely to be appeared in CLI of kubernetes ecology like `kubeadm reset`, etc. 

It only delete Karmada without `force`, and replace `recursive` function with `force`.